tests/test_config.py

```diff
@@ -120,346 +144,382 @@ def check_schema_error(expected_msg):
         """The real checker."""
         with pytest.raises(CraftError) as cm:
             load(tmp_path)
-        assert str(cm.value) == dedent(expected_msg)
+        assert dedent(expected_msg) in str(cm.value)
 
     return check_schema_error
 
 
-def test_schema_top_level_no_extra_properties(create_config, check_schema_error):
+@pytest.mark.parametrize(
```

Comment:
Please make sure each of these tests also tests the version of the charm from the old version. In this case, an additional parameter set that contains the `charmcraft.yaml` from before without having a `metadata.yaml`

Reply:
This is test the old version. The new version is not added in this PR.